README.md

```diff
@@ -396,6 +397,74 @@ end
 
 The Dashboard can be set to automatically refresh by checking "Live Poll" in the Dashboard header, or by setting `?poll=10` with the interval in seconds (default 30 seconds).
 
+### Batches
+
+Batches track a set of jobs, and enqueue an optional callback job when all of the jobs have finished (succeeded or discarded).
+
+- A simple example that enqueues your `MyBatchCallbackJob` after the two jobs have finished, and passes along the current user as a batch property:
+
+    ```ruby
+    GoodJob::Batch.enqueue(MyBatchCallbackJob, user: current_user) do
+      MyJob.perform_later
+      OtherJob.perform_later
+    end
+
+    # When these jobs have finished, it will enqueue your `MyBatchCallbackJob.perform_later(batch, options)`
+    class MyBatchCallbackJob < ApplicationJob
+      def perform(batch, options = {})
+        # Do something with the <GoodJob::Batch> instance; options is a hash is reserved for future use
+        batch.properties[:user] # => <User id: 1, ...>
+      end
+    end
+    ```
+
+- Jobs can be added to an existing batch. Jobs in a batch are enqueued and performed immediately, though the final callback job will not be enqueued until `GoodJob::Batch#enqueue` is called.
```

Comment:
I find these instructions unclear. To me, it looks like the code below contains a race condition. What happens if all the jobs finishes before `enqueue` is called?.

On another note: What happens if you call `enqueue` twice on the same batch?
What happens if you call `enqueue` on a batch without any jobs?

Reply:
Great callout! I will cover that in the documentation. Briefly:

- the callback job won't be enabled until `#enqueue` is called on the batch. 
- calling `#enqueue` on a batch without any jobs will immediately enqueue the callback job
- `#enqueue` can be called multiple times, which will re-enable the callback job to be enqueued
- `#add` will add jobs to the batch without triggering an enqueue

Do you think this kind of example makes sense for creating a multistage batch?

```ruby
class MyBatchJob < ApplicationJob
  def perform(batch, options = {})
    if batch.properties[:stage] == 1
      batch.properties[:stage] = 2
      batch.enqueue do
        10.times { OtherJob.perform_later}
      end
    elsif batch.properties[:stage] == 2
      batch.properties[:stage] = 3
      batch.enqueue do
        OtherJob.perform_later
      end
    else
      # all done
    end 
  end
end

GoodJob::Batch.enqueue(MyBatchJob, stage: 1)
```